evap/staff/views.py

```diff
@@ -311,27 +313,47 @@ def semester_import(request, semester_id):
     semester = get_object_or_404(Semester, id=semester_id)
     raise_permission_denied_if_archived(semester)
 
-    form = ImportForm(request.POST or None, request.FILES or None)
+    excel_form = ImportForm(request.POST or None, request.FILES or None)
 
-    if form.is_valid():
+    errors = []
+    warnings = []
+    success_messages = []
+
+    if request.method == "POST":
         operation = request.POST.get('operation')
         if operation not in ('test', 'import'):
             raise SuspiciousOperation("Invalid POST operation")
 
-        # extract data from form
-        excel_file = form.cleaned_data['excel_file']
-        vote_start_date = form.cleaned_data['vote_start_date']
-        vote_end_date = form.cleaned_data['vote_end_date']
-
         test_run = operation == 'test'
+        import_run = operation == 'import'
 
-        # parse table
-        EnrollmentImporter.process(request, excel_file, semester, vote_start_date, vote_end_date, test_run)
         if test_run:
-            return render(request, "staff_semester_import.html", dict(semester=semester, form=form))
-        return redirect('staff:semester_view', semester_id)
-    else:
-        return render(request, "staff_semester_import.html", dict(semester=semester, form=form))
+            delete_import_file(request.user.id)  # remove old file if still exists
+            excel_form.excel_file_required = True
+            if excel_form.is_valid():
+                excel_file = excel_form.cleaned_data['excel_file']
+                file_content = excel_file.read()
+                success_messages, warnings, errors = EnrollmentImporter.process(file_content, semester, None, None, test_run)
+                if not errors:
+                    save_import_file(excel_file, request.user.id)
+
+        elif import_run:
+            file_content = get_import_file_content_if_exists(request.user.id)
+            if not file_content:
+                raise SuspiciousOperation("No test run performed previously.")
```

Comment:
how about rausing the `SuspiciosOperation` inside `get_import_file_content_if_exists`?